### PR TITLE
Use platform specific line separator in gen-l10n

### LINKED ISSUE #109761: [gen_l10n] Generated files should respect the platform's line separator
When projects are running on the Windows platform, files should be generated with <kbd>CRLF</kbd> rather than <kbd>LF</kbd>.

image

```
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [版本 10.0.22000.856], locale zh-CN)
    • Flutter version 3.0.5 at X:\SDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] IntelliJ IDEA Ultimate Edition (version 2022.2)
    • IntelliJ at X:\IDEs\apps\IDEA-U\ch-0\222.3345.118
    • Flutter plugin version 69.0.5
    • Dart plugin version 222.3345.108
```

### PR DESCRIPTION
**Pull request by thkim1011**
Currently files are not generated with `\r\n` in windows. This PR should fix the issue.

Fixes https://github.com/flutter/flutter/issues/109761.

### CODE REVIEW

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io';
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/localizations/gen_l10n.dart:5]: Don't import `dart:io` into the tool. Instead you can use: https://github.com/christopherfujino/flutter/blob/re-enable-packages-autoroller/packages/flutter_tools/lib/src/base/platform.dart#L38

**Comment by AlexV525** [comment]
One thing we should consider is whether the repo (if it's version-controlled by Git) is configured as CRLF or LF, because there are multiple options when checking out files on Windows, such as "Checkout as-is, Commit as-is" and "Checkout Windows-style, Commit Unix-style".

**Comment by christopherfujino** [comment]
> One thing we should consider is whether the repo (if it's version-controlled by Git) is configured as CRLF or LF, because there are multiple options when checking out files on Windows, such as "Checkout as-is, Commit as-is" and "Checkout Windows-style, Commit Unix-style".

Yeah, it seems to me like there isn't one universally correct solution for all Windows users.

Apparently the command `git ls-files --eol` will list all files in the repo and eol style it uses in the local checkout: https://stackoverflow.com/questions/35801024/how-to-see-what-type-of-line-endings-are-already-in-git-repository.

However, on my windows machine, where I'm pretty sure I have git configured to "Checkout Windows-style, Commit Unix-style", I have a mixture of `w/lf` and `w/crlf`, so idk what's going on.

**Comment by thkim1011** [comment]
@AlexV525 @christopherfujino I changed it so that the generated files uses the same line terminator as that of `pubspec.yaml` which I think is the cleanest way to do this? Thoughts on this?

**Comment by AlexV525** [comment]
> @AlexV525 @christopherfujino I changed it so that the generated files uses the same line terminator as that of `pubspec.yaml` which I think is the cleanest way to do this? Thoughts on this?

Should be great enough IMO. +1

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**